bookkeeper-dist/src/main/resources/LICENSE-all.bin.txt

```diff
@@ -202,507 +202,170 @@
+[24] Source available at https://github.com/prometheus/client_java/tree/parent-0.0.21
+[25] Source available at https://github.com/vert-x3/vertx-auth/tree/3.4.1
+[26] Source available at https://github.com/eclipse/vert.x/tree/3.4.1
+[27] Source available at https://github.com/vert-x3/vertx-web/tree/3.4.1
+[28] Source available at https://github.com/javax-inject/javax-inject/tree/1
+[29] Source available at http://logging.apache.org/log4j/1.2/download.html
+[30] Source available at https://github.com/java-native-access/jna/tree/3.2.7
+[31] Source available at https://git-wip-us.apache.org/repos/asf?p=commons-collections.git;a=tag;h=a3a5ad
+[32] Source available at https://git-wip-us.apache.org/repos/asf?p=commons-lang.git;a=tag;h=3ad2e8
+[33] Source available at https://github.com/apache/zookeeper/tree/release-3.5.3
+[34] Source available at https://github.com/eclipse/jetty.project/tree/jetty-9.4.5.v20170502
 
 ------------------------------------------------------------------------------------
-For lib/com.google.protobuf-protobuf-java-3.4.0.jar
-
-Copyright 2014, Google Inc.  All rights reserved.
-
-Redistribution and use in source and binary forms, with or without
-modification, are permitted provided that the following conditions are
-met:
-
-    * Redistributions of source code must retain the above copyright
-      notice, this list of conditions and the following disclaimer.
-    * Redistributions in binary form must reproduce the above
-      copyright notice, this list of conditions and the following disclaimer
-      in the documentation and/or other materials provided with the
-      distribution.
-    * Neither the name of Google Inc. nor the names of its
-      contributors may be used to endorse or promote products derived from
-      this software without specific prior written permission.
-
-THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
-"AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
-LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
-A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
-OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
-SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
-LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
-DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
-THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
-(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
-OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
-
-Code generated by the Protocol Buffer compiler is owned by the owner
-of the input file used when generating it.  This code is not
-standalone and requires a support library to be linked with it.  This
-support library is itself covered by the above license.
+lib/io.netty-netty-3.10.1.Final.jar bundles some 3rd party dependencies.
```

Comment:
if we are not use the 3rd party dependencies that netty bundled with, we shouldn't include their license and notices. because those dependencies might already be excluded.

Reply:
Removed those that aren't relevant. There's a bunch of the notices which refer to code which netty has copied into their own source repo, so I've retained those in the LICENSE file (they're all BSD/MIT).

Reply:
can you point me links on how those licenses should handle "copied code"? 

My interpretation here is those "copied code" has already been compiled and redistributed as part of the jar, we should just retain netty's notice, but we don't need to their license, no?

Reply:
Retaining the notice is enough if the code is ASLv2. With BSD/MIT, the notice is the license, so the whole thing has to be redistributed.